cardano-api/src/Cardano/Api/LedgerState.hs

```diff
@@ -1412,49 +1422,53 @@ nextEpochEligibleLeadershipSlots sbe sGen serCurrEpochState ptclState
 -- See Leader Value Calculation in the Shelley ledger specification.
 -- We need the certified natural value from the VRF, active slot coefficient
 -- and the stake proportion of the stake pool.
-isLeadingSlots :: forall v era. ()
+isLeadingSlotsTPraos :: forall v. ()
   => Crypto.Signable v Shelley.Spec.Seed
   => Crypto.VRFAlgorithm v
   => Crypto.ContextVRF v ~ ()
-  => HasField "_d" (Core.PParams (ShelleyLedgerEra era)) UnitInterval
-  => ShelleyBasedEra era
-  -> (SlotNo, SlotNo) -- ^ Slot range of interest
+  => Set SlotNo
+  -> PoolId
+  -> Map (SL.KeyHash 'SL.StakePool Shelley.StandardCrypto) (SL.IndividualPoolStake Shelley.StandardCrypto)
   -> Consensus.Nonce
-  -> ProtocolParameters
   -> Crypto.SignKeyVRF v
-  -> Rational -- ^ Stake pool relative stake
   -> Shelley.Spec.ActiveSlotCoeff
-  -> Set SlotNo
-isLeadingSlots sbe (firstSlotOfEpoch, lastSlotofEpoch) eNonce pParams vrfSkey
-             stakePoolStake activeSlotCoeff' =
-  let certified :: SlotNo -> Crypto.OutputVRF v
-      certified s = certifiedNaturalValue s eNonce vrfSkey
-
-      pp :: Core.PParams (ShelleyLedgerEra era)
-      pp = toLedgerPParams sbe pParams
-
-      slotRangeOfInterest :: Set SlotNo
-      slotRangeOfInterest = Set.fromList [firstSlotOfEpoch .. lastSlotofEpoch]
-
-      isLeader :: SlotNo -> Bool
-      isLeader s = not (Ledger.isOverlaySlot firstSlotOfEpoch (getField @"_d" pp) s)
-                 && TPraos.checkLeaderValue (certified s)
-                                            stakePoolStake activeSlotCoeff'
-  in Set.filter isLeader slotRangeOfInterest
- where
-  certifiedNaturalValue
-    :: SlotNo
-    -> Consensus.Nonce
-    -> Crypto.SignKeyVRF v
-    -> Crypto.OutputVRF v
-  certifiedNaturalValue slot epochNonce vrfSkey' =
-   Crypto.certifiedOutput
-   $ Crypto.evalCertified () (TPraos.mkSeed TPraos.seedL slot epochNonce) vrfSkey'
+  -> Either LeadershipError (Set SlotNo)
+isLeadingSlotsTPraos slotRangeOfInterest poolid snapshotPoolDistr eNonce vrfSkey activeSlotCoeff' = do
```

Comment:
Nice simplifications! Thanks for discerning which parts of the Consensus API you could use here 👍